flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -106,21 +93,32 @@ object UserDefinedFunctionUtils {
     val actualSignature = typeInfoToClass(signature)
     val evalMethods = checkAndExtractEvalMethods(function)
 
-    evalMethods
+    val filtered = evalMethods
       // go over all eval methods and find one matching
-      .find { cur =>
+      .filter { cur =>
       val signatures = cur.getParameterTypes
       // match parameters of signature to actual parameters
-      actualSignature.length == signatures.length &&
-        signatures.zipWithIndex.forall { case (clazz, i) =>
-          parameterTypeEquals(actualSignature(i), clazz)
-        }
+      (actualSignature.length >= signatures.length &&
+        actualSignature.zipWithIndex.forall { case (clazz, i) =>
+          (i < signatures.length && parameterTypeEquals(clazz, signatures(i))) ||
+            (i >= signatures.length - 1 && cur.isVarArgs &&
+              parameterTypeEquals(clazz, signatures.last.getComponentType))
+        }) ||
+        // match empty variable arguments
+        (actualSignature.length == signatures.length - 1 && cur.isVarArgs)
```

Comment:
It is not guaranteed that actualSignature.lenth is 0 here as you mentioned in the comment.
So if your eval function looks like 
`eval(String a, int... b)`, and call with `udf(1)`, it will pass the validation incorrectly.

Actually i am suggesting to separate the whole filtering into some major branches. For example to check whether `cur.isVarArgs` first, like
```
filter {
    case cur if !cur. isVarArgs => 
       // do normal function matching
    case cur if cur.isVarArgs =>
       // do matching with consideration of varargs
}
```